src/main/scala/is/hail/expr/Type.scala

```diff
@@ -890,31 +1037,31 @@ final case class TArray(elementType: Type) extends TIterable {
   override def scalaClassTag: ClassTag[IndexedSeq[AnyRef]] = classTag[IndexedSeq[AnyRef]]
 }
 
-final case class TSet(elementType: Type) extends TIterable {
+final case class TSet(elementType: Type, override val required: Boolean = false) extends TIterable {
   val elementByteSize: Long = UnsafeUtils.arrayElementSize(elementType)
 
   val contentsAlignment: Long = elementType.alignment.max(4)
 
   override val fundamentalType: TArray = TArray(elementType.fundamentalType)
```

Comment:
all of these `fundamentalType`s need the `required` flag. I'm not sure we want to abstract this because then we can't get precise type information.